server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/filtering/RuleDTO.java

```diff
@@ -254,13 +316,13 @@ public ActionDTO getAction() {
     }
 
     public Rule toRule() {
-        return Rule.builder()
+        Rule.Builder ruleBuilder = Rule.builder()
             .id(Rule.Id.of(id))
             .name(name)
-            .condition(conditionDTO.toCondition())
-            .name(name)
-            .action(actionDTO.toAction())
-            .build();
+            .conditionGroup(conditionGroupDTO.toConditionGroup())
+            .action(actionDTO.toAction());
+
+        return ruleBuilder.build();
```

Comment:
No need to create a temporary Rule.Builder object, you could return the all struct directly?